bench: fix the upsert benchmark

knz opened the pull request:
Prior to this patch, this benchmark was churning on the client
side for no good reason. This patch alleviates the issue so that
benchmark numbers are more reflective of what happens in
CockroachDB.

Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/21313)
<!-- Reviewable:end -->

knz committed:
bench: fix the upsert benchmark

Prior to this patch, this benchmark was churning on the client
side for no good reason. This patch alleviates the issue so that
benchmark numbers are more reflective of what happens in
CockroachDB.

Release note: none

jordanlewis reviewed (approved):
LGTM!
